### PR TITLE
Integrate meetups

### PR DESCRIPTION
**Pull request by DanielKambich**
Added integration of all major meetup functionality (barring reporting meetups). Attempted the context approach however, ended up doing a promise based approach instead. Tested with Lucas and Ryan and nothing broke and seems to work as intended

### CODE REVIEW

File: src/components/meetups/MeetupDetailsPage.tsx [hunk 1]
```diff
@@ -27,22 +31,38 @@ const styles = StyleSheet.create({
...
+
+  // TODO: Timestamp currently is just an integer, need to convert it to a time
+  const { id, timestamp, duration, location, user_list } = meetup; // eslint-disable-line camelcase
+  // Convert users dict to a list of accepted users
+  const acceptedUsers = [];
+  for (const [key, value] of Object.entries(user_list)) {
+    if (value === 1) {
+      acceptedUsers.push(key);
+    }
+  }
```
Comment (stephend017) [inline src/components/meetups/MeetupDetailsPage.tsx]: maybe we should think about adding a refactor to the meeting card UI to betting illustrate this

File: src/components/meetups/MeetupListPage.tsx [hunk 2]
```diff
@@ -31,10 +39,54 @@ const styles = StyleSheet.create({
...
+
+  // Allows automatic loading on the page when navigating to the page from a nested or parent screen
+  useEffect(() => {
+    const unsubscribe = navigation.addListener("focus", () => {
+      loadMeetupData();
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, [navigation]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (stephend017) [inline src/components/meetups/MeetupListPage.tsx:93]: this looks good but I think its gonna change / conflict with my PR for routing

File: src/contexts/meetupContext.tsx [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+  let venuesAsString = "[";
+  for (let index = 0; index < venues.length; index++) {
+    venuesAsString += `"${venues[index]}",`;
+  }
+  venuesAsString = venuesAsString.substring(0, venuesAsString.length - 1);
+  venuesAsString += "]";
+  console.log(venuesAsString);
+  const endpoint = `${SERVER_ADDRESS}/meeting/${meetupID}/venue?email=${email}&token=${token}&venues=${venuesAsString}`;
+  return await AuthenticatedRequestHandler.put(endpoint, mapMeetupResponseJson);
+}
```
Comment (stephend017) [inline src/contexts/meetupContext.tsx:275]: theses functions look very good, but  you aren't using the dispatch function to update state, this is gonna create a few errors I think, I can help you debug this at the meeting today.

**Review by stephend017 — COMMENTED** [review]
Overall looks very good, lots of great changes here. At the meeting today we need to iron out the upcomming merge conflicts between this PR and my PR #57

File: src/components/meetups/MeetupDetailsPage.tsx (hunk 1 above)
Comment (DanielKambich) [reply src/components/meetups/MeetupDetailsPage.tsx]: I agree, it should be a simple enough refactor to do

File: src/components/meetups/MeetupListPage.tsx (hunk 2 above)
Comment (DanielKambich) [reply src/components/meetups/MeetupListPage.tsx:93]: Sounds good, we will discuss this in the meeting then

File: src/contexts/meetupContext.tsx (hunk 3 above)
Comment (DanielKambich) [reply src/contexts/meetupContext.tsx:275]: That sounds good. In my testing I haven't ran into any errors, but also technically these aren't relying on a global state, rather we call and get server updates when loading the screen

**Pull request merged**